### PR TITLE
ENH: Add sr option to actions that use Minimap2

### LINKED ISSUE #27: ENH: expose `sr` option
minimap2 has a short read mode (`sr`) that makes it more efficient for processing short read data. 

Some of the actions in this plugin (e.g., for search, classification, filtering) should have this option exposed in case users input short read data (e.g., Illumina reads).

### PR DESCRIPTION
**Pull request by ChristosMatzoros**
In general, Minimap2 is optimized for long reads (ONT, PacBio, etc.). By using the `sr` option, we can achieve better performance when aligning short reads (e.g., Illumina).

This PR:

- Adds the `sr` option to the preset parameters for all actions that use Minimap2.

- Adds all the missing preset options for `minimap2-search` and `classify_consensus_minimap2`.

This PR closes #27.

### Set up the environment
 ```shell
mamba create -n q2-minimap2 -c conda-forge -c bioconda -c https://packages.qiime2.org/qiime2/2024.2/shotgun/passed/ -c defaults q2cli q2-types q2-feature-classifier minimap2 bs4 samtools gzip chopper nanoplot
```

###  Activate q2-minimap2 environment
```shell
conda activate q2-minimap2
```

### Clone the repo and checkout the PR branch
```bash
git clone https://github.com/bokulich-lab/q2-minimap2.git
cd q2-minimap2
git fetch origin pull/30/head:pr-30
git checkout pr-30
pip install -e .
```

### 1. Download Dataset
[here](https://polybox.ethz.ch/index.php/s/NnWZX2r6vgf0nAh)

### 2. Test it out!

Using `sr` option with single-end reads:
```bash
qiime minimap2 filter-reads --i-query-reads single-end.qza  --i-index-database human_index_sr.qza --p-mapping-preset sr --o-filtered-query-reads mapped_se.qza --verbose
```

Using `sr` option with paired-end reads:
```bash
qiime minimap2 filter-reads --i-query-reads paired-end.qza --i-index-database human_index_sr.qza --p-mapping-preset sr --o-filtered-query-reads mapped_pe.qza --verbose
```

Using default preset `map-ont` option with paired-end reads:
```bash
qiime minimap2 filter-reads --i-query-reads paired-end.qza  --i-index-database human_index_sr.qza --o-filtered-query-reads mapped_pe.qza --verbose
```

### CODE REVIEW

File: q2_minimap2/_action_params.py [hunk 1]
```diff
@@ -8,22 +8,43 @@
...
     PairedEndSequencesWithQuality,
     SequencesWithQuality,
 )
 from q2_types.sample_data import SampleData
-from qiime2.plugin import Bool, Choices, Float, Int, Range, Str, TypeMatch
+from qiime2.plugin import Bool, Choices, Float, Int, Range, Str, TypeMap, TypeMatch
 
 from q2_minimap2.types._type import Minimap2IndexDB, PairwiseAlignmentMN2
 
 T = TypeMatch([SequencesWithQuality, PairedEndSequencesWithQuality])
```
Comment (Sann5) [inline q2_minimap2/_action_params.py:20]: Is this still needed?

**Review by Sann5 — APPROVED** [review]
Looks good to me @ChristosMatzoros! The tests pass and the actions compleate without errors (except for the last one in which we expected one). Just a comment. In the captured output I get a warning:

```bash
[WARNING] Indexing parameters (-k, -w or -H) overridden by parameters used in the prebuilt index.
```

Is this expected?

**Comment by ChristosMatzoros** [comment]
Thank you @Sann5 !

I will have a better look at why you get the warning, maybe I missed it. If I create the index using the `sr` option then I think I shouldn't get this warning when I use the `filter-reads`. Basically it says that different flags were used during building the index from the ones I give when mapping. 
I will investigate it before I merge!

**Comment by ChristosMatzoros** [comment]
Hey @Sann5,

I made two changes:

- It seems that it is possible to use `paired-end` reads with the `sr` option in Minimap2, so I removed the code for the `SampleData[JoinedSequencesWithQuality] `input where I used the `TypeMap`. Now everything works as it should for both `single-end` and `paired-end` reads.

- The warning you got was because I used both the `preset` option and the `kmer_length` option when indexing. What the preset does is essentially provide many parameters together for Minimap2. For example, for `sr`: `(-k21 -w11 --sr --frag=yes -A2 -B8 -O12,32 -E2,1 -b0 -r100 -p.5 -N20 -f1000,5000 -n2 -m20 -s40 -g100 -2K50m --heap-sort=yes --secondary=no)`. When both options are used, one of the options (in this case, the kmer length `-k`) is overritten. This means it will not have the value based on the sr preset. So, if I use `minimap2 -k 15 -x sr ...`, Minimap2 will set the `kmer length` to `15`. Then, when performing the alignment and using the `sr` option, Minimap2 will notice that the value of `k` does not match the one it should have for `sr` `(=21)` but instead has the overridden value set during indexing `(=15)`.

I have renewed the datasets, and you can rerun the analysis if you want. Now you shouldn't get the warning, and you should also be able to use `paired-end` reads with the `sr` option.

Let me know what you think,
Thank you!

**Review by Sann5 — COMMENTED** [review]
I still get a warning with the last command @ChristosMatzoros. Is this expected?

**Comment by ChristosMatzoros** [comment]
Hey @Sann5 ,

Sorry, I forgot to add it in the description. Yes, in the last one it is expected because we use different preset during indexing (sr) and different during the mapping process (`map-ont` which is the default).

**Pull request merged**